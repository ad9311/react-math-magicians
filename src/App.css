@import url('https://fonts.googleapis.com/css2?family=Montserrat:wght@400;500;600&display=swap');

body {
  margin: 0;
  padding: 0;
  color: hsl(0, 5%, 25%);
  font-family: 'Montserrat', sans-serif;
}

h1 {
  margin: 0;
}
