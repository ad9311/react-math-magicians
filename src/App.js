import React from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import Navbar from './pages/Navbar';
import Home from './pages/Home';
import CalculatorPage from './pages/CalculatorPage';
import Quote from './pages/Quote';
import './App.css';

const App = () => (
  <Router>
    <Navbar />
    <Switch>
      <main className="pcon-6">
        <Route exact path="/">
          <Home />
        </Route>
        <Route path="/calculator">
          <CalculatorPage />
        </Route>
        <Route path="/quote">
          <Quote />
        </Route>
      </main>
    </Switch>
  </Router>
);

export default App;
