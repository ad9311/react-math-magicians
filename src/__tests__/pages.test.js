import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import NavBarMock from '../mocks/navbarMocks';
import App from '../App';

describe('Test routes', () => {
  test('test home', () => {
    render(<App />);
    fireEvent.click(screen.getByText('Home'));
    expect(screen.getByText('Welcome to our page!')).toMatchSnapshot();
  });

  test('test calculator', () => {
    render(<App />);
    fireEvent.click(screen.getByText('Calculator'));
    expect(screen.getByText('Lets do some math!')).toMatchSnapshot();
  });

  test('test quotes', () => {
    render(<App />);
    fireEvent.click(screen.getByText('Quote'));
    expect(screen.getByText('William Paul Thurston.')).toMatchSnapshot();
  });
});

describe('test navBar', () => {
  test('nav home', () => {
    render(<NavBarMock />);
    expect(screen.getByText('Home')).toMatchSnapshot();
  });

  test('nav Calculate', () => {
    render(<NavBarMock />);
    expect(screen.getByText('Calculator')).toMatchSnapshot();
  });
  test('nav Quotes', () => {
    render(<NavBarMock />);
    expect(screen.getByText('Quotes')).toMatchSnapshot();
  });
});
