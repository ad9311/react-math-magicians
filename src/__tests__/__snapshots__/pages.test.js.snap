// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test routes test calculator 1`] = `
<h2
  class="fs-4"
>
  Lets do some math!
</h2>
`;

exports[`Test routes test home 1`] = `
<h2
  class="fs-4"
>
  Welcome to our page!
</h2>
`;

exports[`Test routes test quotes 1`] = `
<strong>
   William Paul Thurston.
</strong>
`;

exports[`test navBar nav Calculate 1`] = `
<a
  href="/Calculator"
>
  Calculator
</a>
`;

exports[`test navBar nav Quotes 1`] = `
<a
  href="/Quotes"
>
  Quotes
</a>
`;

exports[`test navBar nav home 1`] = `
<a
  href="/"
>
  Home
</a>
`;
